codes/classical/bits/hamming.yml

```diff
@@ -0,0 +1,40 @@
+#######################################################
+## This is a code entry in the error correction zoo. ##
+##       https://github.com/errorcorrectionzoo       ##
+#######################################################
+
+code_id: hamming
+physical: bits
+logical: bits
+
+name: 'Hamming code'
+#introduced: '\cite{doi: 10.1002/j.1538-7305.1950.tb00463.x.}'
+
+description: 'Hamming Codes are a family of linear codes with parameters \((2^r-1,2^r-r-1, 3)\) for any
+ \(r\),which can detect one-bit and two-bit errors, or correct one-bit errors. The \(r \times (2^r-1) \) parity check matrix,
+ \(H\), has all possible non-zero r-bit strings as its columns. The codewords are \({x|Hx=0}\).'
+
+protection: 'Can detect 1-bit and 2-bit errors, and can correct 1-bit errors'
+
+features:
+  rate: '\(1-\frac{\log n}{n}\)'
+
+realizations:
+  - 'Commonly used when error rates are very low, for example, computer RAM.'
+
+notes:
+  - 'Has the maximum rate among binary codes for that block length and with minimum distance 3.
+  Therefore, it is a perfect code, i.e, it saturates the Hamming Bound.'
+
+relations:
+  parents:
+    - code_id: 'binary_linear'
+      detail: ''
+
+  cousins:
+    - code_id: hadamard
```

Comment:
See above, might not be correct

Reply:
Wiki says the dual of hamming is the shortened hadamard code.. should I correct this to shortened hadamard, and change the hadamard entry to 'extended hamming'?

Reply:
Keep hadamard as cousin. Add detail entry saying the dual of hamming is shortened version of hadamard.